app/controllers/trips_controller.rb

```diff
@@ -1,17 +1,33 @@
 class TripsController < ApplicationController
-  before_action :set_trip, only: [:dashboard, :show, :edit, :update, :destroy]
+  before_action :set_trip, only: [:dashboard, :import, :show, :edit, :update, :destroy]
+  skip_before_action :verify_authenticity_token, only: [:import]
 
   # GET /trips
   # GET /trips.json
   def index
     authorize Trip
-    @trips = Trip.all
-  end
-
+    @trips = Trip.all end
   def dashboard
     authorize Trip
   end
 
+  def import
+    if request.post?
+      authorize Trip
```

Comment:
Can move `authorize Trip` to the top of the method so you don't have to put it twice